### PR TITLE
Remove the ability to write API to a file, use stdout instead

### LINKED ISSUE #10: Allow output to console rather than to disk
There might be cases where a client would like to handle the output themselves and do something different. So we should allow the ability to not provided a save path, and instead just output to console.

### PR DESCRIPTION
**Pull request by ollieatkinson**
Why:
It's far much easier to let the user redirect stdout to a file instead.
Can remove some error handling from the app.
There might be usecases where the end user woould like to use the output
of guise in a different tool.

This change addresses the need by:
Remove writeToFile and replace with fputs

Fix #10

### CODE REVIEW

File: Sources/GuiseFramework/CLI/GenerateCommand.swift [hunk 1]
```diff
@@ -15,27 +15,15 @@ struct GenerateCommand: CommandProtocol {
...
     
-    let path: String
     let noDocumentation: Bool
     
-    static func create(path: String) -> (_ noDocumentation: Bool) -> Options {
-      return { noDocumentation in
-        return self.init(path: path, noDocumentation: noDocumentation)
-      }
+    static func create(noDocumentation: Bool) -> Options {
+      return self.init(noDocumentation: noDocumentation)
```
Comment (paulsamuels) [inline Sources/GuiseFramework/CLI/GenerateCommand.swift]: Can `self` be inferred? `return .init(noDocumentation: Bool)...`

File: Sources/GuiseFramework/CLI/GenerateCommand.swift [hunk 2]
```diff
@@ -77,14 +65,10 @@ private extension GenerateCommand {
...
-  func writeToFile(options: Options) -> (_ source: String) -> Result<Void, APIGeneratorError> {
+  func printToStdout() -> (_ source: String) -> Result<Void, APIGeneratorError> {
     return { source in
-      
-      if options.path.isEmpty {
-        return .failure(.invalidArgument(description: "path must be provided"))
-      } else {
-        return Result(attempt: { try source.write(toFile: options.path, atomically: true, encoding: .utf8) }).mapError({ .failedToWrite(path: options.path, error: $0) })
-      }
+      fputs(source, stdout)
```
Comment (paulsamuels) [inline Sources/GuiseFramework/CLI/GenerateCommand.swift]: Should we check the result of `fputs` - at the moment we are always returning success which might not be the case

File: Sources/GuiseFramework/CLI/GenerateCommand.swift [hunk 3]
```diff
@@ -77,14 +65,10 @@ private extension GenerateCommand {
     }
   }
   
-  func writeToFile(options: Options) -> (_ source: String) -> Result<Void, APIGeneratorError> {
+  func printToStdout() -> (_ source: String) -> Result<Void, APIGeneratorError> {
```
Comment (paulsamuels) [inline Sources/GuiseFramework/CLI/GenerateCommand.swift]: This currying doesn't seem to be necessary

```
  func printToStdout(_ source: String) -> Result<Void, APIGeneratorError> {
    fputs(source, stdout)
    return .success(())
  }

---

.flatMap(printToStdout)
```

**Review by paulsamuels — APPROVED** [review]
Couple of style things to address but makes sense to me 🚀

File: Sources/GuiseFramework/CLI/GenerateCommand.swift (hunk 2 above)
Comment (ollieatkinson) [reply Sources/GuiseFramework/CLI/GenerateCommand.swift]: Yep,

```swift
  func printToStdout(source: String) -> Result<Void, APIGeneratorError> {
    if fputs(source, stdout) == EOF {
      return .failure(.failedToPrint)
    } else {
      return .success(())
    }
  }
```

**Pull request merged**